### PR TITLE
fix: invalid examples/documentation in parquet_derive doc

### LINKED ISSUE #5687: parquet_derive: invalid examples/documentation
**Describe the bug**
There are references to `SerializedFileWriter::close_row_group()` in the following files, which is a method that doesn't seem to exist, rendering the examples un-compileable:
- https://github.com/apache/arrow-rs/blob/11450ae8ddf902b57cb42491a3d824d9550a05ea/parquet_derive/README.md?plain=1#L76
- https://github.com/apache/arrow-rs/blob/11450ae8ddf902b57cb42491a3d824d9550a05ea/parquet_derive/src/lib.rs#L72

### PR DESCRIPTION
**Pull request by Weijun-H**
# Which issue does this PR close?

Closes #5687

# Rationale for this change
 

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by tustvold** [comment]
Perhaps we could change `ignore` to `no_run` to ensure these examples are kept up to date?

File: parquet_derive/src/lib.rs [hunk 1]
```diff
@@ -160,6 +167,7 @@ pub fn parquet_record_writer(input: proc_macro::TokenStream) -> proc_macro::Toke
 ///
 /// pub fn read_some_records() -> Vec<ACompleteRecord> {
 ///   let mut samples: Vec<ACompleteRecord> = Vec::new();
+///   let mut file = tempfile().unwrap();
```
Comment (tustvold) [inline parquet_derive/src/lib.rs:170]:
```suggestion
///   let mut file = todo!();
```
And same above, would avoid a dependency

Comment (Weijun-H) [reply parquet_derive/src/lib.rs]: It does not work.
```rust
error[E0277]: the trait bound `(): std::io::Write` is not satisfied
   --> parquet_derive/src/lib.rs:73:19
    |
34  |  let mut writer = SerializedFileWriter::new(file, schema, Default::default()).unwrap();
    |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `std::io::Write` is not implemented for `()`
    |
note: required by a bound in `SerializedFileWriter`
   --> /Users/hwj/Desktop/rust/arrow-rs/parquet/src/file/writer.rs:141:36
    |
141 | pub struct SerializedFileWriter<W: Write> {
    |                                    ^^^^^ required by this bound in `SerializedFileWriter`
```

Comment (Weijun-H) [reply parquet_derive/src/lib.rs]: I use `std::fs::File` to avoid a dependency

File: parquet_derive/src/lib.rs [hunk 2]
```diff
@@ -160,6 +167,7 @@ pub fn parquet_record_writer(input: proc_macro::TokenStream) -> proc_macro::Toke
 ///
 /// pub fn read_some_records() -> Vec<ACompleteRecord> {
 ///   let mut samples: Vec<ACompleteRecord> = Vec::new();
+///   let mut file = File::open("some_file.parquet").unwrap();
```
Comment (tustvold) [inline parquet_derive/src/lib.rs:170]: I suspect the mut is unnecessary

File: parquet_derive/src/lib.rs [hunk 3]
```diff
@@ -62,14 +66,15 @@ mod parquet_field;
 ///       a_str: "I'm false"
 ///     }
 ///   ];
+///  let mut file = File::open("some_file.parquet").unwrap();
```
Comment (tustvold) [inline parquet_derive/src/lib.rs:69]: I suspect the mut is unnecessary

**Review by tustvold — APPROVED** [review]
Just a minor nit

**Pull request merged**